src/main/java/org/generationcp/middleware/service/impl/dataset/DatasetServiceImpl.java

```diff
@@ -163,6 +166,14 @@ public long countObservationsByInstance(final Integer datasetId, final Integer i
 			//PLOTDATA
 			factorColumns = this.daoFactory.getDmsProjectDAO()
 				.getObservationSetVariables(observationSetId, PLOT_COLUMNS_FACTOR_VARIABLE_TYPES);
+			//STOCK ID
+			final LotsSearchDto lotsSearchDto = new LotsSearchDto();
+			lotsSearchDto.setPlantingStudyIds(Arrays.asList(studyId));
+			if (this.daoFactory.getLotDao().countSearchLots(lotsSearchDto) > 0) {
+				final Optional<MeasurementVariable>
+					designation = factorColumns.stream().filter(measurementVariable -> measurementVariable.getName().equals("DESIGNATION")).findFirst();
```

Comment:
Is it better to compare by Id in this case instead of DESIGNATION?

Reply:
Done.